### PR TITLE
Added check for local storage

### PR DESCRIPTION
**Pull request by WillMcGill**
Added a check for local storage.  If the info is stored locally, populate state based on cached info.
If no local storage is available.  Pull data from API.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -7,29 +7,39 @@ import axios from 'axios';
...
+      })
+    }
+    else {
+      this.state.map(MenuItems => {
+        axios.get('https://entree-f18.herokuapp.com/v1/menu/12')
+          .then(items => {
+            localStorage.setItem(MenuItems.name, JSON.stringify(items.data.menu_items))
+            this.state.Menu = items.data.menu_items;
+          })
+          ;
```
Comment (kevin-moore-729) [inline src/App.js]: The semicolon on line 38 should probably come immediately after the closing parenthesis on line 37

File: src/App.js [hunk 2]
```diff
@@ -7,29 +7,39 @@ import axios from 'axios';
...
+
+    // Check if menu data already exists in local storage.
+    //If so, store data in local storage in menu state, else pull from API
+
+    if (localStorage.getItem('Breakfast')) {
+      this.state.map(MenuItems_Local => {
+        this.state.Menu = JSON.parse(localStorage.getItem(MenuItems_Local.name));
+      })
+    }
+    else {
```
Comment (kevin-moore-729) [inline src/App.js]: Your `else {` should probably follow the `}` on line 30

File: src/App.js [hunk 3]
```diff
@@ -7,29 +7,39 @@ import axios from 'axios';
...
+      this.state.map(MenuItems => {
+        axios.get('https://entree-f18.herokuapp.com/v1/menu/12')
+          .then(items => {
+            localStorage.setItem(MenuItems.name, JSON.stringify(items.data.menu_items))
+            this.state.Menu = items.data.menu_items;
+          })
+          ;
+      })
+
+
```
Comment (kevin-moore-729) [inline src/App.js]: You could probably get rid of the empty lines on 40-41

File: src/App.js [hunk 4]
```diff
@@ -7,29 +7,39 @@ import axios from 'axios';
...
-   
+
+  componentDidMount() {
+
+    // Check if menu data already exists in local storage.
+    //If so, store data in local storage in menu state, else pull from API
+
+    if (localStorage.getItem('Breakfast')) {
+      this.state.map(MenuItems_Local => {
+        this.state.Menu = JSON.parse(localStorage.getItem(MenuItems_Local.name));
```
Comment (kevin-moore-729) [inline src/App.js:28]: You should use `this.setState` whenever you're changing a value in your component state. I also don't think `this.state.Menu` is the value you want to change here.

Comment (kevin-moore-729) [reply src/App.js:28]: `MenuItems_Local.Menu`

File: src/App.js (hunk 1 above)
Comment (WillMcGill) [reply src/App.js]: Done. Semicolon moved to line 37

File: src/App.js (hunk 2 above)
Comment (WillMcGill) [reply src/App.js]: Done. Else moved to line 30

File: src/App.js (hunk 3 above)
Comment (WillMcGill) [reply src/App.js]: Done, Lines 40 and 41 deleted

File: src/App.js (hunk 4 above)
Comment (WillMcGill) [reply src/App.js:28]: This is very helpful to know.  I am actually removing the Menu objects from the state object and will be referencing the local storage directly depending on the page.

**Comment by WillMcGill** [comment]
Thanks @kevin-moore-729.  Very helpful info.  @jhalljhall

**Pull request merged**